src/ext-handler.cpp

```diff
@@ -176,38 +179,95 @@ void SeafileExtensionHandler::stop()
     }
 }
 
-void SeafileExtensionHandler::generateShareLink(const QString& repo_id,
-                                                const QString& path_in_repo,
-                                                bool is_file,
-                                                bool internal)
+void SeafileExtensionHandler::getShareLink()
 {
     // qDebug("path_in_repo: %s", path_in_repo.toUtf8().data());
-    const Account account = seafApplet->accountManager()->getAccountByRepo(repo_id);
+    const Account account = seafApplet->accountManager()->currentAccount();
     if (!account.isValid()) {
         return;
     }
 
-    if (internal) {
-        QString path = path_in_repo;
-        if (!is_file && !path.endsWith("/")) {
+    if (shared_link_req_params_.internal) {
+        QString path = shared_link_req_params_.path_in_repo;
+        if (!shared_link_req_params_.is_file && !path.endsWith("/")) {
             path += "/";
         }
-        SeafileLinkDialog *dialog = new SeafileLinkDialog(repo_id, account, path, NULL);
+        SeafileLinkDialog *dialog = new SeafileLinkDialog(
+            shared_link_req_params_.repo_id, account, path, NULL);
         dialog->setAttribute(Qt::WA_DeleteOnClose);
         dialog->show();
         dialog->raise();
         dialog->activateWindow();
     } else {
         GetSharedLinkRequest *req = new GetSharedLinkRequest(
-            account, repo_id, path_in_repo, is_file);
+            account, shared_link_req_params_.repo_id,
+            shared_link_req_params_.path_in_repo);
+
+        connect(req, SIGNAL(success(const SharedLinkInfo&)),
+                this, SLOT(getShareLinkSuccess(const SharedLinkInfo&)));
+        connect(req, SIGNAL(failed()),
+                this, SLOT(generateShareLink()));
+
+        req->send();
+    }
+}
+
+void SeafileExtensionHandler::getShareLinkSuccess(const SharedLinkInfo& shared_link_info)
+{
+    bool proceed = false;
+    proceed = seafApplet->detailedYesOrNoBox(tr("<b>Warning:</b> The shared link already exists, delete and create link anyway?"),
```

Comment:
The existing link is also checked in `AdvancedSharedLinkDialog::checkPreviousSharedLink()`, why do that in two places?